svcb.go

```diff
@@ -669,6 +673,62 @@ func (s *SVCBIPv6Hint) copy() SVCBKeyValue {
 	}
 }
 
+// SVCBDoHPath pair is used to indicate the URI template that the
+// clients may use to construct a DNS over HTTPS URI.
+//
+// See RFC xxxx (https://datatracker.ietf.org/doc/html/draft-ietf-add-svcb-dns-02)
+// and RFC yyyy (https://datatracker.ietf.org/doc/html/draft-ietf-add-ddr-06).
+//
+// Basic use pattern for using the dohpath option together with an alpn
+// option:
+//
+//	s := new(dns.SVCB)
+//	s.Hdr = dns.RR_Header{Name: ".", Rrtype: dns.TypeSVCB, Class: dns.ClassINET}
+//	e := new(dns.SVCBAlpn)
+//	e.Alpn = []string{"h2", "h3"}
```

Comment:
I need Alpn because? (haven't read the drafts)

Reply:
The two RFC drafts that define or use `dohpath` ([`draft-ietf-add-svcb-dns`][1] and [draft-ietf-add-ddr-06][2]) both use these two together.  Neither of them says that the `dohpath` parameter cannot be used without the `alpn` one *explicitly,* but the first of the draft reads:

 >  If the "alpn" SvcParamKey indicates support for HTTP, "dohpath" MUST be present \[…\]

And the semantics of using `dohpath` without the `alpn` set to one of the HTTP ones seem to remain unclear.

Since the code illustrates a “basic use pattern”, which seems to basically always use `alpn` and `dohpath` together, I've decided to make the example use those together as well.

[1]: https://datatracker.ietf.org/doc/html/draft-ietf-add-svcb-dns-02
[2]: https://datatracker.ietf.org/doc/html/draft-ietf-add-ddr-06.html